Pinta.Core/Effects/ColorMapping.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Pinta.Core;
+
+public abstract class ColorMapping
```

Comment:
I'm not really clear on the split here between `ColorMapping` and `ColorGradient`, and if it's necessary to have the base class at all? In particular, having factory methods here which create a `ColorGradient` feels wrong for a base class

Reply:
There could be color mappings that are not gradients, hence this structure. But I could get rid of the `abstract` class

Reply:
I don't necessarily mind having the separate `ColorMapping` to allow for other implementations, but if so I think things should be less tightly coupled.
e.g.
- Just make `ColorMapping` an interface and move anything else to `ColorGradient`
- Should the effects only work with a `ColorMapping` and leave it to the effect dialog to create a gradient?